pkg/telemetry/client.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package telemetry
+
+import (
+	"net"
+	"os"
+	"time"
+
+	proto "github.com/golang/protobuf/proto"
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+)
+
+const (
+	DefaultInterval = time.Hour
+
+	// MTUs are often 1500, though a good argument could be made for 512
+	DefaultPacketSize = 1000
+)
+
+// ClientOpts allows you to set Client Options
+type ClientOpts struct {
+	// Interval is how frequently stats from the provided Registry will be
+	// sent up. Note that this interval is "jittered", so the actual interval
+	// is taken from a normal distribution with a mean of Interval and a
+	// variance of Interval/4. Defaults to DefaultInterval
+	Interval time.Duration
+
+	// Application is the application name, usually prepended to metric names.
+	// By default it will be os.Args[0]
+	Application string
+
+	// Instance is a string that identifies this particular server. Could be a
+	// node id, but defaults to the first non-nil MAC address
+	Instance string
+
+	// PacketSize controls how we fragment the data as it goes out in UDP
+	// packets. Defaults to DefaultPacketSize
+	PacketSize int
+
+	// Registry is where to get stats from. Defaults to monkit.Default
+	Registry *monkit.Registry
+}
+
+// Client is a telemetry client for sending UDP packets at a regular interval
+// from a monkit.Registry
+type Client struct {
+	opts     ClientOpts
+	addr     string
+	packet   MetricPacket
+	buf      proto.Buffer
+	overhead int
+}
+
+// NewClient constructs a telemetry client that sends packets to remote_addr
+// over UDP.
+func NewClient(remote_addr string, opts ClientOpts) (rv *Client, err error) {
```

Comment:
I think this could benefit/be simplified by using the functional options pattern:

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis